java/api/src/main/java/com/exadel/frs/core/trainservice/dao/SubjectDao.java

```diff
@@ -51,18 +49,23 @@ public Subject deleteSubjectByName(final String apiKey, final String subjectName
 
     @Transactional
     public int removeAllSubjectEmbeddings(final String apiKey, final String subjectName) {
+        List<Subject> subjectList = new ArrayList<>();
         final Optional<Subject> subjectOptional = subjectRepository.findByApiKeyAndSubjectNameIgnoreCase(apiKey, subjectName);
         if (subjectOptional.isEmpty()) {
             // nothing has been removed
-            return 0;
+            subjectList = subjectRepository.findByApiKey(apiKey);
+        } else {
+            subjectList.add(subjectOptional.get());
         }
 
-        final var subject = subjectOptional.get();
-
-        int deleted = embeddingRepository.deleteBySubjectId(subject.getId());
```

Comment:
If I understand correctly, the "deleted" variable contains a number of embeddings deleted.
But you in 65 line count number of subjects deleted.
Let's return the number of embeddings deleted always.

Reply:
done

Reply:
Sorry, do not see any changes